libsolidity/interface/StandardCompiler.cpp

```diff
@@ -490,7 +490,16 @@ std::optional<Json::Value> checkOptimizerDetailSteps(Json::Value const& _details
 				);
 			}
 
-			_setting = _details[_name].asString();
+			if (
+				auto const delimiterPos = _details[_name].asString().find(":");
+				delimiterPos != string::npos
+			)
+			{
+				_optimiserSetting = _details[_name].asString().substr(0, delimiterPos);
+				_cleanupSetting = _details[_name].asString().substr(delimiterPos + 1);
+			}
+			else
+				_optimiserSetting = _details[_name].asString();
```

Comment:
Maybe in this case, this is actually shorter and nicer :-):
```suggestion
			string fullSequence = _details[_name].asString();
			auto const delimiterPos = fullSequence.find(":");
			_optimiserSetting = fullSequence.substr(0, delimiterPos);
			if (delimiterPos != string::npos)
				_cleanupSetting = fullSequence.substr(delimiterPos + 1);
```
We could still eliminate one